services/src/main/java/io/grpc/services/BinaryLog.java

```diff
@@ -416,10 +416,10 @@ static Peer socketToProto(SocketAddress address) {
   static io.grpc.binarylog.Metadata metadataToProto(Metadata metadata, int maxHeaderBytes) {
     Preconditions.checkState(maxHeaderBytes >= 0);
     Builder builder = io.grpc.binarylog.Metadata.newBuilder();
-    if (maxHeaderBytes > 0) {
-      byte[][] serialized = InternalMetadata.serialize(metadata);
+    // This code is tightly coupled with Metadata's implementation
+    byte[][] serialized;
+    if (maxHeaderBytes > 0 && (serialized = InternalMetadata.serialize(metadata)) != null) {
```

Comment:
suggestion: I'd add a null check for ```metadata``` before setting ```serialized ```

Reply:
Good idea